src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/versionTracker.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.hmpps.architecture.documentation
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.structurizr.model.Container
+import uk.gov.justice.hmpps.architecture.cloneRepository
+import java.io.File
+
+data class Version(
+  val softwareSystem: String,
+  val application: String,
+  val applicationUrl: String,
+  val circleciOrbVersion: String,
+  val gradleBootPluginVersion: String,
+  val chartVersions: String,
+)
+
+fun parseVersions(containersWithGit: List<Container>): List<Version> {
+  return containersWithGit
+    .also { println("[parseVersions] Found ${it.size} applications with github repos") }
+    .mapNotNull { parseVersion(it) }
+}
+
+private fun parseVersion(container: Container): Version? {
+  val r = cloneRepository(container) ?: return null
+  return Version(
+    softwareSystem = container.softwareSystem.name,
+    application = container.name,
+    applicationUrl = container.url,
+    circleciOrbVersion = readCircleOrbVersion(r),
+    gradleBootPluginVersion = readGradlePluginVersion(r),
+    chartVersions = readHelmChartDependencies(r)
+  )
+}
+
+private fun readCircleOrbVersion(repo: File): String {
+  val circleConfig = repo.resolve(".circleci").resolve("config.yml").takeIf { it.exists() }?.readText().orEmpty()
+
+  val circleOrb = Regex("ministryofjustice/(hmpps@[\\d.]*)").find(circleConfig)?.groups?.get(1)?.value
+  val dpsOrb = Regex("ministryofjustice/(dps@[\\d.]*)").find(circleConfig)?.groups?.get(1)?.value
+  return listOfNotNull(circleOrb, dpsOrb).joinToString("<br>")
+}
+
+const val GRADLE_PATTERN = """id\("uk.gov.justice.hmpps.gradle-spring-boot"\) version "([^"]*)""""
+private fun readGradlePluginVersion(repo: File): String {
+  val buildFileKt = repo.resolve("build.gradle.kts").takeIf { it.exists() }?.readText().orEmpty()
+  val buildFile = repo.resolve("build.gradle").takeIf { it.exists() }?.readText().orEmpty()
+
+  return listOfNotNull(
+    Regex(GRADLE_PATTERN).find(buildFileKt)?.groups?.get(1)?.value,
+    Regex(GRADLE_PATTERN).find(buildFile)?.groups?.get(1)?.value,
+  ).joinToString("<br>")
+}
+
+private fun readHelmChartDependencies(repo: File): String {
+  val chartFile = repo.resolve("helm_deploy")
+    .walkTopDown().find { it.name == "Chart.yaml" }
+    ?: return "no helm chart"
+
+  val dependencies = ObjectMapper(YAMLFactory()).readTree(chartFile).get("dependencies")
+    ?: return "standalone chart"
+
+  return dependencies.joinToString("<br>") { "${it.get("name").textValue()}@${it.get("version").textValue()}" }
```

Comment:
Might be worth adding a bullet to each dependency?

Reply:
Will try

Reply:
Well, we'll have to -- structurizr ignores the <br> tags
<img width="1918" alt="image" src="https://user-images.githubusercontent.com/1526295/143027827-d91b4118-4e34-4045-a81a-480cf7bdecb1.png">